share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,854 @@
+## Wake Tutorial
+
+This tutorial will teach you the basics of Wake language. After completeing this tutorial, you will have typed and executed many basic wake functions.
+
+### Invoking wake
+
+1. Create a folder inside `/scratch` directory. `mkdir /scratch/<your_name>/wake_tutorial`.
+
+2. `cd` into the directory. `cd /scratch/<your_name>/wake_tutorial`.
+
+3. Create a wake database: `wake --init .`.After this you will see a `wake.db` file created inside the folder.
+
+4. Create a file with `.wake` extension and copy the functions mentioned in this tutorial and run using `wake` command in the terminal as mentioned in Step 5.  
+
+5. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`. Ex: `wake -x 'double 2'`.
+   
+   `v` stands for verbosity and `x` stands for execute the expression.
+
+6. To know about various wake options run `wake --help`. 
+
+7. For more information on wake library click [here](https://sifive.github.io/wake/).
+
+### wake `Hello World`
+
+Copy the below function into a `.wake` file you created in step 4
+```
+def abc = "Hello World"
+```
+Then execute the following command 
+```
+wake -x 'abc'
+```
+This will call the function `abc` and print `Hello World`.
+
+* You can also do this without a `.wake` file by doing it all in one command: `wake -x '"Hello World"'`
+
+### Defining Functions
+A function should be defined using the keyword `def`. Simple function definition takes the form
+`def <function name> <parameter list> = <expression>`
+
+We cannot specify explicitly the return type and type of the parameters. When the function body is of multi-line the return value is specified by the last line in the function.
+
+Functions can be defined with the keyword `global` which makes the function to be accessed outside the file where it is being defined.
```

Comment:
```suggestion
Functions can be defined with the keyword `global` which makes the function to be accessed from other .wake files other than the one where it is defined.
```

Reply:
when it is defined as `global` you can use those functions or the values by calling them in other wake files as well.